mesheryctl/internal/cli/root/registry/update.go

```diff
@@ -69,27 +78,26 @@ mesheryctl registry update --spreadsheet-id 1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdw
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-
-		srv, err := mutils.NewSheetSRV(spreadsheeetCred)
-		if err != nil {
-			utils.Log.Error(ErrUpdateRegistry(err, modelLocation))
-			return err
-		}
-		resp, err := srv.Spreadsheets.Get(spreadsheeetID).Fields().Do()
-		if err != nil || resp.HTTPStatusCode != 200 {
-			utils.Log.Error(ErrUpdateRegistry(err, outputLocation))
-			return err
+		if csvDir != "" {
+			utils.Log.Info("Using local CSV directory: ", csvDir)
+			err := InvokeComponentsUpdateFromCSV()
+			if err != nil {
+				utils.Log.Error(err)
+				return err
+			}
+			return nil
 		}
 
-		sheetGID = GetSheetIDFromTitle(resp, "Components")
-
-		err = InvokeCompUpdate()
-		if err != nil {
-			utils.Log.Error(err)
+		if spreadsheeetID != "" && spreadsheeetCred != "" {
+			utils.Log.Info("Using Google Sheet with ID: ", spreadsheeetID)
+			err := InvokeComponentsUpdateFromSheets()
+			if err != nil {
+				utils.Log.Error(err)
+				return err
+			}
 			return nil
 		}
-
-		return nil
+		return errors.New("please provide a Google Sheet or a local csv directory")
```

Comment:
Because the logic before and after parsing the local csv and google sheets is the same, I think it's better to extract the parsing part only, instead of the whole `InvokeComponentsUpdate()`. Another check can be added initially like
```suggestion
	parsedComponents := map[string]map[string][]utils.ComponentCSV{}
		if csvDir != "" {
			utils.Log.Info("Using local CSV directory: ", csvDir)
			parsedComponents, err := parseLocalCSVDirectory(csvDir)
			if err != nil {
				utils.Log.Error(err)
				return err
			}
			return nil
		}

		if spreadsheeetID != "" && spreadsheeetCred != "" {
			utils.Log.Info("Using Google Sheet with ID: ", spreadsheeetID)
			parsedComponents, err := parsedGoogleSheets()
			if err != nil {
				utils.Log.Error(err)
				return err
			}
			return nil
		}

		InvokeComponentsUpdate(parsedComponents)
```